### PR TITLE
crypto: introduce references to crypto IC

### PR DESCRIPTION
**Pull request by ldts**
Initial introduction to crypto IC and the API.
The documentation should grow with dedicated sections to each IC.

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

### CODE REVIEW

File: architecture/crypto.rst [hunk 1]
```diff
@@ -169,12 +170,37 @@ are the main things to consider when adding a new crypto provider:
...
+device. This session is shared with the normal world through the PKCS#11
+interface but also through a more generic interface (`libseetec`_) which allows
+clients to send Application Protocol Data Units (APDUs) directly to the device.
+
+Notice that cryptographic co-processors do not necessarily comply with all the
+GP requirements tested and covered by the OP-TEE sanity test suite
+(`optee_test`_). In those cases where the cryptographic operations are not
+supported - i.e: the SE05x does not implement all RSA key sizes - we opted for
+disabling those particular tests at build time rather than letting them fail.
+
```
Comment (jenswi-linaro) [inline architecture/crypto.rst:193]: Empty line not needed.

**Review by jenswi-linaro — COMMENTED** [review]
One minor comment. With that fixed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by ldts** [comment]
@jenswi-linaro tag added and comment fixed

File: architecture/crypto.rst (hunk 1 above)
Comment (ldts) [reply architecture/crypto.rst:193]: um, without that empty line, the hyperlinks break

Comment (jenswi-linaro) [reply architecture/crypto.rst:193]: Fair enough, then it's needed.

**Comment by ldts** [comment]
@jbech-linaro ok to merge?

**Comment by jbech-linaro** [comment]
@ldts , yes, looking good, I'll merge it now, thanks for the contribution!

**Pull request merged**